src/main/scala/com/github/sbt/sbom/BomExtractor.scala

```diff
@@ -189,6 +191,125 @@ class BomExtractor(settings: BomExtractorParams, report: UpdateReport, log: Logg
     }
   }
 
+  private def purl(group: String, name: String, version: String): String =
+    new PackageURL(PackageURL.StandardTypes.MAVEN, group, name, version, new util.TreeMap(), null).canonicalize()
+
+  private def dependencies: Seq[Dependency] = {
+    val dependencies = configurationsForComponents(settings.configuration).flatMap { configuration =>
+      dependenciesForConfiguration(configuration)
+    }.distinct // deduplicate dependencies reported by multiple configurations
+
+    dependencies
+  }
+
+  private def dependenciesForConfiguration(configuration: Configuration): Seq[Dependency] = {
+    report
+      .configuration(configuration)
+      .toSeq
+      .flatMap { configurationReport =>
+        log.info(
+          s"Configuration name = ${configurationReport.configuration.name}, details: ${configurationReport.details.size}"
+        )
+
+        new DependenciesExtractor(configurationReport).dependencies
+      }
+  }
+
+  class DependenciesExtractor(configurationReport: ConfigurationReport) {
+    def dependencies: Seq[Dependency] =
+      moduleGraph.nodes
+        .sortBy(_.id.idString)
+        .map { node =>
+          val bomRef = purl(node.id.organization, node.id.name, node.id.version)
+
+          val dependency = new Dependency(bomRef)
+
+          val dependsOn = moduleGraph.dependencyMap.getOrElse(node.id, Nil).sortBy(_.id.idString)
+          dependsOn.foreach { module =>
+            val bomRef = purl(module.id.organization, module.id.name, module.id.version)
+            dependency.addDependency(new Dependency(bomRef))
+          }
+
+          dependency
+        }
+
+    // https://github.com/sbt/sbt/blob/1.10.x/main/src/main/scala/sbt/internal/graph/backend/SbtUpdateReport.scala
```

Comment:
Hmm. Sorry to go back-and-forth on this a bit, but I just noticed that this is including Apache-licensed code into an MIT-licensed project. That might be possible (they're both premissively-licensed, after all), but we'd want to double-check that adding this comment is sufficient, or that we'd want to do a bit more to make sure this is responsibly tracked.

Reply:
That makes sense, thank you for checking this as well